### PR TITLE
Sign PSBT view for signing outside nodeguard

### PR DESCRIPTION
**Pull request by RodriFS**
To be merged after https://github.com/Elenpay/Nodeguards-Companion/pull/11

This PR is about a view to directly paste a PSBT into the extension and sign it with one of the keys, rather than making the extension detect the psbt in the page. This is because sometimes we want to sign a PSBT that is not in nodeguard and the user has to set up Sparrow instead

### CODE REVIEW

**Comment by daliclovr** [comment]
What is this about? @RodriFS

**Comment by RodriFS** [comment]
> What is this about? @RodriFS

Sorry, I'll update the PR

File: extension/src/features/approve_pasted_psbt.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+#[function_component(ApprovePastedPSBT)]
+pub fn approve_pasted_psbt() -> Html {
+    let password = use_context::<UserContext>()
+        .unwrap()
+        .password
+        .clone()
+        .unwrap_or_default();
+    let navigator = use_navigator().unwrap();
+    let storage = UserStorage::read(LocalStorage::default());
```
Comment (daliclovr) [inline extension/src/features/approve_pasted_psbt.rs]: You use `LocalStorage::default()` lots of times, what about creating it once at the beginning and reusing it across? 

`let local_storage = LocalStorage::default();`

and then you clone it when you need it if you need to pass the ownership

File: extension/src/features/approve_pasted_psbt.rs [hunk 2]
```diff
@@ -0,0 +1,136 @@
+use crate::{
+    components::{
+        select::{Select, SelectItem},
+        textarea::TextArea,
+    },
+    context::UserContext,
+    utils::{helpers::get_clipboard, storage::LocalStorage},
+};
+use anyhow::anyhow;
+use anyhow::Result;
```
Comment (daliclovr) [inline extension/src/features/approve_pasted_psbt.rs]: You can group them!

File: extension/src/features/approve_pasted_psbt.rs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                .get_wallet_mut(&selected_wallet_value)
+                .ok_or_else(|| anyhow!("Wallet not found"))
+                .and_then(|wallet| {
+                    decode_psbt_and_sign(&psbt, wallet, &password, settings_storage.get_network())
+                        .map_err(|e| anyhow!("Error while signing PSBT {e}"))
+                });
+
+            match result {
+                Ok(p) => signed_psbt.set(p),
+                Err(e) => error.set(format!("{e}")),
```
Comment (daliclovr) [inline extension/src/features/approve_pasted_psbt.rs]: I think you you can do: `e.to_string()` instead

File: extension/src/features/approve_pasted_psbt.rs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        })
+    };
+
+    let onclick_save = {
+        let selected_wallet_value = selected_wallet_value.clone();
+        let psbt = psbt.clone();
+        Callback::from(move |_: MouseEvent| {
+            if password.is_empty() {
+                return;
+            }
```
Comment (daliclovr) [inline extension/src/features/approve_pasted_psbt.rs:53]: does the user know that the password is empty? how does yew process this? asks the user for a password again?

**Review by daliclovr — DISMISSED** [review]

**Comment by daliclovr** [comment]
@RodriFS left a couple of comments to improve code readability, but functionally its ok to me :)

File: extension/src/features/approve_pasted_psbt.rs (hunk 1 above)
Comment (RodriFS) [reply extension/src/features/approve_pasted_psbt.rs]: Moved to RefCell 💯

File: extension/src/features/approve_pasted_psbt.rs (hunk 2 above)
Comment (RodriFS) [reply extension/src/features/approve_pasted_psbt.rs]: done

File: extension/src/features/approve_pasted_psbt.rs (hunk 3 above)
Comment (RodriFS) [reply extension/src/features/approve_pasted_psbt.rs]: yes 😄

File: extension/src/features/approve_pasted_psbt.rs (hunk 4 above)
Comment (RodriFS) [reply extension/src/features/approve_pasted_psbt.rs:53]: Button is disabled

**Review by daliclovr — APPROVED** [review]

**Pull request merged**